enterprise/frontend/src/embedding-sdk/types/index.ts

```diff
@@ -1,17 +1,28 @@
 import type { JSX } from "react";
 
 import type { FetchRequestTokenFn } from "embedding-sdk";
-import type { SdkErrorProps } from "embedding-sdk/components/private/PublicComponentWrapper/SdkError";
+import type { SdkErrorComponent } from "embedding-sdk/store/types";
 
-export type SDKConfig = {
+type BaseSDKConfig = {
   metabaseInstanceUrl: string;
-  jwtProviderUri: string;
   loaderComponent?: () => JSX.Element;
-  errorComponent?: ({ message }: SdkErrorProps) => JSX.Element;
+  errorComponent?: SdkErrorComponent;
 
   /**
    * Specifies a function to fetch the refresh token.
    * The refresh token should be in the format of { id: string, exp: number }
    */
   fetchRequestToken?: FetchRequestTokenFn;
 };
+
+export type SDKConfigWithJWT = BaseSDKConfig & {
```

Comment:
extended the type to hopefully make type-narrowing easier in `use-init-data`

Reply:
These types look good to me. Especially the way we use `never`